### PR TITLE
Add script to deploy web app

### PR DESCRIPTION
**Pull request by belcherj**
### Fix
Losing GitHub Actions we made the call to make deployment local rather than
adding another level of CI

### Test
1. Checkout branch
2. `npm run deploy staging`
3. Does it push to webapp-staging?
4. Does staging.

### Review
Only one developer is required to review these changes, but anyone can perform the review.

### Release
 - Added script to deploy web app [#1723](https://github.com/Automattic/simplenote-electron/pull/1723)

### CODE REVIEW

File: bin/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  *)
+    exit 0
+    ;;
+esac
+
+# Install Node Modules
+npm install
+
+# Temporarily move config-local so it doesn't get overwritten
+mv config-local.json config-local.origional.json
```
Comment (codebykat) [inline bin/deploy.sh]: typo for "original"

File: bin/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+# Clean up old builds
+rm -rf dist
+
+# Build site
+NODE_ENV=production make build
+
+# Restore config-local
+rm config-local.json
+mv config-local.origional.json  config-local.json
```
Comment (codebykat) [inline bin/deploy.sh]: "original"

File: bin/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+# Checkout deploy branch
+if [ "$(git branch | grep $COMMIT_BRANCH)" ]
+then
+  git branch -D $COMMIT_BRANCH
+fi
+git checkout -b $COMMIT_BRANCH
+
+# Delete files not needed in production
+find . -maxdepth 1 ! -name '.git' ! -name 'dist' ! -name 'vip' -exec rm -rf {} \;
```
Comment (codebykat) [inline bin/deploy.sh]: It looks like this find is deleting the config .json files

File: bin/deploy.sh (hunk 1 above)
Comment (belcherj) [reply bin/deploy.sh]: fixed

File: bin/deploy.sh (hunk 3 above)
Comment (belcherj) [reply bin/deploy.sh]: fixed

File: bin/deploy.sh [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+# Copy vip files to base path
+cp -r vip/* ./
+rm -rf vip
+
+# Add, commit, push
+touch .gitignore
+echo config-local.json >> .gitignore
+git add --all
+git commit -m "Build: $COMMIT_BRANCH [ci skip]"
```
Comment (dmsnell) [inline bin/deploy.sh]: if we're wanting to deploy a given branch why are we building this and making a commit to that branch?
do we not have a way to inject a different config for the different environments?

seems like we ought to be able to do it like this…

```bash
git checkout webapp-develop
git reset --hard develop
```

Does something in the build process push itself out to VIP Go?

Comment (belcherj) [reply bin/deploy.sh]: When the head of the webapp, webapp-staging, and webapp-develop branches change VIP Go grabs the contents of the repo and pushes it to the server.

We do not have a way to inject a different config for the different environments.

I am not sure what the code suggestion achieves.

Comment (dmsnell) [reply bin/deploy.sh]: the code suggestion would change the `HEAD` of those branches and should trigger web hooks to fire. the branches effectively would work like tags.

Comment (belcherj) [reply bin/deploy.sh]: VIP Go doesn't build the app it only grabs what's in the folder. https://github.com/Automattic/simplenote-electron/tree/webapp

**Pull request merged**